src/test/java/io/appium/java_client/youiengine/frames/YouiEngineAppiumSampleApp.java

```
@@ -0,0 +1,88 @@
+package io.appium.java_client.youiengine.frames;
+
+import io.appium.java_client.youiengine.YouiEngineDriver;
+import io.appium.java_client.youiengine.util.TestUtility;
+import org.openqa.selenium.NoSuchElementException;
+
+/**
+ * This class is used to encapsulate all common application behaviours and make them
+ * available to test authors from one object. Test scripts instance this object and call methods
+ * from this class or its member objects.
+ *
+ * By wrapping interactions in methods based off of UI components, a hierarchy is created that
+ * allows test authors to write simple code. The actual code behind the interactions can be
+ * updated in one place which detaches the test scripts from the implementation reducing
+ * maintenance should the application under test change.
+ *
+ * Created by paulfoster on 2016-07-21.
+ */
+
+public class YouiEngineAppiumSampleApp {
+    private YouiEngineDriver driver;
+    private static TestUtility utility;
+
+    public LanderFrame landerScreen;
+    public TextEditFrame textEditScreen;
+    public ButtonsFrame buttonsScreen;
+
+    /* Constructor takes an instance of the YouiEngineDriver. This ensures there is an open and
+         running session to use and allows the test author to swap capabilities at the test script
+         level.
+
+         This allows the YouiEngineSampleApp class to be platform independent.   */
+    public YouiEngineAppiumSampleApp(YouiEngineDriver youiEngineDriver) {
+        driver = youiEngineDriver;
+        landerScreen = new LanderFrame(driver);
+        textEditScreen = new TextEditFrame(driver);
+        buttonsScreen = new ButtonsFrame(driver);
+    }
+
+    /* Helper methods to allow test authors to perform app level navigation. The leverage common
+         designs in the app such as the presence of a Back button on the second tier screens. */
+    public void goHome() {
+        try {
+            driver.findElementByName("Btn-Back").click();
```

I didn't really want to throw an exception on this call because the app is simple - if you're on the main screen there will not be a back button. If you're on a subscreen there will be. I'm using this pattern intentionally.

Another thought is checking to see if the Text Edit button exists, it'd be more certain and not an assumption that we're on the lander.

I agree, it's likely better to use exists first.

```
if (!exists("Text Edit) {
   if (!exists("Btn-Back")) {
      throw new NoIdeaWhatScreenYouAreOnException("I made this up");
   }
   driver.findElementByName("Btn-Back").click();
}
```

Would work?